jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -477,12 +477,19 @@ def _image_viewer_update(self, viewer, x, y):
                     dq_value = dq_data[int(round(y)), int(round(x))]
                 unit = image.get_component(attribute).units
             elif isinstance(viewer, CubevizImageView):
+                skip_spectral_density_eqv = False
                 arr = image.get_component(attribute).data
                 unit = image.get_component(attribute).units
                 value = self._get_cube_value(
                     image, arr, x, y, viewer
                 )
-                if self.image_unit is not None:
+
+                # We don't want to convert for things like moment maps
+                if str(u.Unit(unit).physical_type) not in ("spectral flux density",
+                                                           "surface brightness"):
+                    skip_spectral_density_eqv = True
```

Comment:
if this is the only exception, this probably could be shortened to be directly in the if-statement itself, but that's probably a matter of preference.